google/cloud/internal/async_read_write_stream_auth.h

```diff
@@ -20,6 +20,7 @@
 #include "google/cloud/version.h"
 #include <functional>
 #include <memory>
+#include <thread>
```

Comment:
is this for `std::mutex`?  Why not just `<mutex>`?

Reply:
Because I don't know the threading headers by heart yet :) done.